test.js

```diff
@@ -7,668 +7,465 @@ const Next = require('next')
-    t.equal(res.headers['content-type'], 'text/html; charset=utf-8')
-  })
+  fastify.close()
 })
 
-test('should support a custom handler', t => {
-  t.plan(3)
+test('should support a custom handler', async t => {
+  const fastify = await Fastify().register(require('./index'))
+  fastify.next('/hello', (app, req, reply) => {
+    app.render(req.raw, reply.raw, '/hello', req.query, {})
+  })
 
-  const fastify = Fastify()
-  t.teardown(() => fastify.close())
+  const origin = await fastify.listen({ port })
+  const { statusCode, headers } = await request({ path: '/hello', origin })
 
-  fastify
-    .register(require('./index'))
-    .after(() => {
-      fastify.next('/hello', (app, req, reply) => {
-        app.render(req.raw, reply.raw, '/hello', req.query, {})
-      })
-    })
+  t.equal(statusCode, 200)
+  t.equal(headers['content-type'], 'text/html; charset=utf-8')
 
-  fastify.inject({
-    url: '/hello',
-    method: 'GET'
-  }, (err, res) => {
-    t.error(err)
-    t.equal(res.statusCode, 200)
-    t.equal(res.headers['content-type'], 'text/html; charset=utf-8')
-  })
+  fastify.close()
 })
 
-test('should return 404 on undefined route', t => {
-  t.plan(2)
+test('should return 404 on undefined route', async t => {
+  const fastify = await Fastify().register(require('./index'))
+  fastify.next('/hello')
 
-  const fastify = Fastify()
-  t.teardown(() => fastify.close())
+  const origin = await fastify.listen({ port })
+  const { statusCode } = await request({ path: '/test', origin })
 
-  fastify
-    .register(require('./index'))
-    .after(() => {
-      fastify.next('/hello')
-    })
+  t.equal(statusCode, 404)
 
-  fastify.inject({
-    url: '/test',
-    method: 'GET'
-  }, (err, res) => {
-    t.error(err)
-    t.equal(res.statusCode, 404)
-  })
+  fastify.close()
 })
 
-test('should throw if path is not a string', t => {
-  t.plan(2)
+test('should throw if path is not a string', async t => {
+  const fastify = await Fastify().register(require('./index'))
 
-  const fastify = Fastify()
-  fastify
-    .register(require('./index'))
-    .after(err => {
-      t.error(err)
-      try {
-        fastify.next(null)
-        t.fail()
-      } catch (e) {
-        t.equal(e.message, 'path must be a string')
-      }
-    })
+  try {
+    fastify.next(null)
 
-  fastify.close()
+    t.fail()
+  } catch (e) {
+    t.equal(e.message, 'path must be a string')
+  }
 })
 
-test('should throw if opts.method is not a string', t => {
-  t.plan(2)
+test('should throw if opts.method is not a string', async t => {
+  const fastify = await Fastify().register(require('./index'))
 
-  const fastify = Fastify()
-  fastify
-    .register(require('./index'))
-    .after(err => {
-      t.error(err)
-      try {
-        fastify.next('/hello', { method: 1 })
-        t.fail()
-      } catch (e) {
-        t.equal(e.message, 'options.method must be a string')
-      }
-    })
+  try {
+    fastify.next('/hello', { method: 1 })
```

Comment:
cant we instead of the try catch block and the fallback t.fail() call t.throws? would make the test easier to digest imho

```suggestion
    t.throws(() =>  fastify.next('/hello', { method: 1 }), 'options.method must be a string')
```

Reply:
I prefer the suggested approach too, because it's way too easy to forget about having that t.fail() in place, meaning that if no error is thrown, the test won't fail

Reply:
I like it, I'll commit a change doing this for the other try/catch tests too. Thanks! BTW I'd accept this commit suggestion, but Standard is complaining of `Multiple spaces found before 'fastify'. (no-multi-spaces)`

Reply:
Done. Please check again and resolve if you're ok with the changes.